cmd/resource-migrator/main.go

```diff
@@ -0,0 +1,212 @@
+// The resource migrator allow us to migrate all the resources saved on disk to a storage provider such as S3.
+package main
+
+import (
+	"context"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+	res "github.com/gazebo-web/fuel-server/bundles/common_resources"
+	"github.com/gazebo-web/fuel-server/bundles/models"
+	"github.com/gazebo-web/fuel-server/bundles/worlds"
+	"github.com/gazebo-web/fuel-server/globals"
+	"github.com/gazebo-web/fuel-server/vcs"
+	"github.com/gazebo-web/gz-go/v7"
+	"github.com/gazebo-web/gz-go/v7/storage"
+	"github.com/jinzhu/gorm"
+	_ "github.com/jinzhu/gorm/dialects/mysql"
+	"github.com/schollz/progressbar/v3"
+	"log"
+	"os"
+	"os/signal"
+	"strconv"
+	"syscall"
+	"time"
+)
+
+const (
+	maxParallelUploads = 15
+)
+
+func main() {
+	s, db := setup()
+	defer gz.Close(db)
+
+	run(s, db)
+}
+
+func run(s storage.Storage, db *gorm.DB) {
+	started := time.Now()
+
+	// Get the list of resources to upload for: Models
+	ms, modelListSize, err := getResourceToUpload[*models.Model](db)
+	if err != nil {
+		log.Panicln("Failed to get models:", err)
+	}
+
+	// Get the list of resources to upload for: Worlds
+	ws, worldListSize, err := getResourceToUpload[*worlds.World](db)
+	if err != nil {
+		log.Panicln("Failed to get worlds:", err)
+	}
+
+	// Define progress bar
+	c := make(chan uploadRequest, (modelListSize+worldListSize)*10)
```

Comment:
```suggestion
	// Channel to request resource uploads
	c := make(chan uploadRequest, (modelListSize+worldListSize)*10)
```

Also, any specific reason why the channel size is 10 times the number of resources?

Reply:
Yes, thanks for raising this comment. each model and world could have multiple versions, I used 10 as an arbitrary value to catch all possible scenarios without iterating over the list of elements to upload.

Reply:
Now that I've realized that different versions are not being handled by uploadRequests, this doesn't make sense.